utils/zerovec/src/ule/error.rs

```diff
@@ -2,31 +2,47 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use core::any;
 use core::fmt;
 
 /// A generic error type to be used for decoding slices of ULE types
 #[derive(Copy, Clone, Debug)]
-pub enum ULEError<E> {
+pub enum ULEError {
     InvalidLength { ty: &'static str, len: usize },
-    ParseError(E),
+    ParseError { ty: &'static str },
+    FormatError,
```

Comment:
Suggestion: It looks like this variant is only used when validating a VZV; correct?  If so, a more descriptive name might be useful, since otherwise people might be inclined to return this from their own ULE impls.  Alternatively, consider using ParseError with an appropriate `ty` instead of having a separate variant for this case.